src/vivarium_gates_mncnh/components/observers.py

```diff
@@ -80,3 +88,102 @@ def register_observations(self, builder: Builder) -> None:
                 COLUMNS.PREGNANCY_OUTCOME,
             ],
         )
+
+
+class MaternalDisordersBurdenObserver(Observer):
+    @property
+    def configuration_defaults(self) -> dict[str, Any]:
+        return {
+            "stratification": {
+                self.get_configuration_name(): {
+                    "exclude": [],
+                    "include": [],
+                    "data_sources": {
+                        f"{cause}_ylds": partial(self.load_ylds_per_case, cause=cause)
+                        for cause in self.maternal_disorders
+                    },
+                },
+            },
+        }
+
+    def __init__(self):
+        super().__init__()
+        self.maternal_disorders = MATERNAL_DISORDERS
+
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+
+    def get_configuration(self, builder):
+        return builder.configuration["stratification"][self.get_configuration_name()]
+
+    def register_observations(self, builder: Builder) -> None:
```

Comment:
This class needs to do the following that is reflected below:
-record number of cases for each maternal disorder
-record number of deaths for each maternal disorder
-record ylls and ylds for each maternal disorder